cms/djangoapps/contentstore/views/tabs.py

```diff
@@ -110,13 +110,60 @@ def update_tabs_handler(course_item: CourseBlock, tabs_data: Dict, user: User) -
     """
 
     if "tabs" in tabs_data:
-        reorder_tabs_handler(course_item, tabs_data, user)
+
+        # if the first tab in the requested tabs to be re ordered is a static tab
+        # then reordering would be for static tabs alone. Static tabs are alone
+        first_tab = get_tab_by_tab_id_locator(course_item.tabs, tabs_data["tabs"][0])
+        if isinstance(first_tab, StaticTab):
+            reorder_static_tabs_handler(course_item, tabs_data, user)
+        else:
+            reorder_tabs_handler(course_item, tabs_data, user)
```

Comment:
I don't understand the need for this complexity. 

The original code could already handle reordering two static tabs, and that was all that was needed. I think even if you leave this code untouched it will work.

What issue is this change solving?

Reply:
@xitij2000 when the tabs are reordered with the original code, we get an `InvalidTabsException`. See in the image below.


![Screenshot 2021-09-16 at 11 24 34](https://user-images.githubusercontent.com/30666999/133597377-309aafa1-799a-4ab5-a2ca-09ada33c9ceb.png)

Now that we've updated the Custom Pages design, in the POST request only the static tabs are given. Because of that the `requested_tab_id_locators` list [here](https://github.com/edx/edx-platform/blob/2845786a140e5bb6246f8a4ba068650e589b7dd0/cms/djangoapps/contentstore/views/tabs.py#L129) only contains static tabs. So when the `new_tab_list` is created [here](https://github.com/edx/edx-platform/blob/2845786a140e5bb6246f8a4ba068650e589b7dd0/cms/djangoapps/contentstore/views/tabs.py#L135) it puts all the static tabs first and appends all the other tabs which causes the `InvalidTabsException` [here](https://github.com/edx/edx-platform/blob/2845786a140e5bb6246f8a4ba068650e589b7dd0/common/lib/xmodule/xmodule/tabs.py#L482)

Originally the tabs in the POST request came with the other tabs that were non static, so we didn't have this issue.

The new `reorder_static_tabs_handler` takes care of this by rearranging the `new_tab_list` as expected by the validation process. I created a new method instead of modifying the previous one assuming that there were other cases that the non static tabs were also reordered. But if only static tabs can be reordered then I can rename `reorder_static_tabs_handler` to `reorder_tabs_handler` and have just one method. The other alternative is to remove this checks [here](https://github.com/edx/edx-platform/blob/2845786a140e5bb6246f8a4ba068650e589b7dd0/common/lib/xmodule/xmodule/tabs.py#L498-L504) in the validation process.

Reply:
The ability to reorder non-static tabs is being removed, so if you need to make changes, you can clean up the code to support doing only that.

Reply:
@xitij2000 [this view](https://github.com/edx/edx-platform/blob/94378cb06c16ef45c778fd2e646bae3c5372c588/cms/djangoapps/contentstore/rest_api/v0/views/tabs.py#L166-L239) reorders course tabs using the `reorder_tabs_handler` method. Can you confirm that it should cater to only static tabs?

Reply:
@tinumide Yes, that is a new API I created recently for this, and it needs to follow the same featureset as the old API. So we can remove everything but static tab handling from both.